### DISCUSSION TITLE
Problem: using ZbStatus3 <device name> with MQTT

### DISCUSSION THREAD

**Discussion by JoeWiseman** [arendst/Tasmota#11980] [Q&A] [answered]
Hi there,

I've bought a Sonoff-Zigbee-Brigde ans flashed it successfully with the latest zigbee2tasmota software.
It should wor together with:

    node red (Redmatic the latest version)
    Mosquito-MQTT broker
    both running on a homematic CCU3

I get the values in a JSON-String by subscribing to /Zigbee/EG-1/tele/SENSOR, where all sensors are combined in one object :-(

Allthough I know that this is more a client-server behaviour I like to read some sensor-values by using the MQTT-command:
/Zigbee/EG-1/cmnd/ZbStatus3 Arbeitszimmer_Sensor

When using the command ZbStatus3 Arbeitszimmer_Sensor in the tasmota-sonsole all works fine,
when using the MQTT-node from node red the console gets an error:
MQT: /Zigbee/EG-1/stat/RESULT = {"Command":"Unknown"}
Even when using the short-address or the number of the device the error occurs....

Is this command not availible via MQTT or is there something else to be considered

Would be great to get some help.

Thanks in advance
Joe

**Comment by sfromis** [comment, marked as answer]
You just need to use MQTT appropriately. Split so the command name is in the topic, and arguments in the payload. Or use Backlog as the command name in the topic, and your full command as the payload.

**Reply by JoeWiseman** [reply]
Thanks for answering!

I've been using an inject node, which is injecting the payload (in my case a string: Arbeitszimmer_Sensor) into an MQTT node with the topic /Zigbee/EG-1/cmnd/ZbStatus3 .... so this should be correct, isn't it?

In the meanwhile I also build a flow which extracts the right data with some other change- and switch-nodes ... nothing really complex ...

... BUT ...

It would be easier to use the /cmnd/ZbStatus3 <devicename> so that I get the senso-data I like to examine without spltting modification ... 

So I wonder IF THERE IS THE POSSIBILITY TO USE ZbStatus3-COMMAND IN MQTT....

Regards
Joe

**Reply by sfromis** [reply]
Absolutely yes. Just as I already described, of course make sure to use the correct topic. Example could be 
```
msg.topic="cmnd/zbbridge-one/Backlog"
msg.payload="ZbStatus3 Ikea-Opal"
```
Topic depends a lot on how you've configured your device, the format I used is in default setup. If you want to check what your Tasmota bridge subscribes to, you can activate `Weblog 3` and restart. Then the console will show messages detailing the topics expected.

Obviously, use the MQTT out and in nodes as usual for the traffic. You may want to make the MQTT in node format the received payloads directly into objects, instead of getting the raw JSON.

**Reply by sfromis** [reply]
Let me add one consideration. I consider it "less practical" to use ZbStatus3 at all, when the payloads about what happens already arrive on the MQTT subscription without having to bother with publishing commands.

**Reply by JoeWiseman** [reply]
Thanks a lot for answering in detail! That helped me out ...
I'm just a bit confused, why it is nesseccary to use the Backlog.... but what's really more important is the fact that I have to subscribe to a "common topic" (for me: /Zigbee/EG-1/stat/RESULT) to get the "answer". Though there are more devices connected to the zigbee-bridge they would all "answer" by using this topic ... So I have to find out which device is actually answering. 
By doing so there will be no advantage to the envisaged method in which I use the tele/SENSOR.

So in the end I thank you for opening my eyes an let me see the light ;-)
Best regards
Joe

**Reply by sfromis** [reply]
It is not necessary to use Backlog. As I wrote "Split so the command name is in the topic, and arguments in the payload". Backlog was just a potentially less confusing way of doing it.
```
msg.topic="cmnd/zbbridge-one/ZbStatus3"
msg.payload="Ikea-Opal"
```
The publish/subscribe model is just how MQTT works. It is an inherently asynchronous design. That's why I added my consideration about just not at all doing commands to poll values, as you already get them as it happens.

**Reply by sfromis** [reply]
Note: You may *not* need to use a common topic to subscribe, there are options for [separated topics](https://tasmota.github.io/docs/Zigbee/#sending-sensor-values-to-separated-mqtt-topics).

**Reply by JoeWiseman** [reply]
Yes I allready have been using the "rule" option, but I decided not to "spread the configuration", so I just added the zigbee-devices to the bridge and give them friendly names ;-)

Again thanks for your answer and the information given!!
